### ISSUE TITLE
How to properly create MethodReference for definition in the same module?

### ISSUE THREAD

**Issue by pamidur** [jbevain/cecil#639]
Hi, I've been struggling to do one thing, that I think is really easy, but I can figure out how.

So in my case I have a type:
```c#
interface ITest<TT>{
    void Do<TM>(){}
}

class MyClass : ITest<int>
{
    // method I want to create
    void TestTarget<int>.Do<TM>(){}
}
```

I need to call method ```Do``` from within the same module. For this I need a reference to method ```ITest.Do``` so I can set its ```DeclaringType=new GenericInstanceType(  <ITest>, <int>   )``` and then pass the result to ```new GenericInstanceMethod()``` and  ```Method.Overrides.Add()```

I tried:
1. Create a reference by utilizing ```Module.ImportReference()``` - doesn't work - returns the same definition

2. Create a reference manually ```new MethodReference()``` and **copy** all the stuff - sorta works, but here when I copy GenericParameters references they are changed in the way that they now belong to this reference instead of original definition which cause exception on final write

3. Create a reference manually ```new MethodReference()``` and **clone** all the stuff - sorta works unless I try to use resulting reference in ```Method.Overrides.Add()```, because it seems like here I need to have a reference with original ```TM``` generic param and not a clone.

So to recap, I couldn't find a proper way to create MethodReference for definition in the same module with original GenericParameter, so I can use it in ```Method.Overrides.Add()```

**Comment by jbevain** [comment]
Hi! Sorry I missed this, please open a discussion item if you're still having issues creating the proper MethodReference.
